django/send_notifications.py

```diff
@@ -0,0 +1,52 @@
+
+from asapsports.db.users import insert_user, get_user_by_asap_token, get_user_by_fb_id, get_user_by_id
+from asapsports.db.notifications import get_current_notification, delete_notification
+from asapsports.db.user_in_game import get_users
+from asapsports.db.games import get_game
+#from utils import get_connection
+from datetime import datetime
+import requests
+import psycopg2
+
+from asapsports.settings import *
+try:
+    from asapsports.local_settings import * # Override with local settings if there are any
+except ImportError:
+    print("Import Error")
+
+#print(locals())
+
+DB_INFO = DATABASES['default']
+
+def get_connection():
+    # Copy the other code for get connection...
+    return psycopg2.connect(dbname=DB_INFO['NAME'],
+                            user=DB_INFO.get('USER'),
+                            password=DB_INFO.get('PASSWORD'),
+                            host=DB_INFO.get('HOST'),
+                            port=DB_INFO.get('PORT'))
+
+notifications = []
+conn = get_connection()
+
+while 1:
+    cur_time = datetime.utcnow().timestamp()
+    notifications = get_current_notification(conn, cur_time)
+    for game_id in notifications:
+        delete_notification(conn, game_id)
+        print(game_id)
+        game = get_game(conn, game_id)
+        message = "You have a " + game.sport + " game in 1 hour"
+        params={
+            "to": "ExponentPushToken[ApS2k4O61aNJiiarjotwjw]",
+            "sound": "default",
+            "body": message
+        }
+        user_ids = get_users(conn, game_id)
+        for user_id in user_ids:
+            print(user_id)
+            user = get_user_by_id(conn, user_id)
+            token = user.push_token    
+            print("here")
```

Comment:
yea good call

Reply:
yup switched the first part just want a little clarification on the deleting dependent on the status of the request because currently the notifications are for game_ids and requests are for user_ids

Reply:
ie could have one user request fail and the rest successful and then this would lead to duplicate notifications for some or people missing out on notifications in the case we dont rollback